src/DataProtection/Extensions/src/DataProtectionProvider.cs

```diff
@@ -24,10 +24,7 @@ public static class DataProtectionProvider
     /// applications on the machine.</param>
     public static IDataProtectionProvider Create(string applicationName)
     {
-        if (string.IsNullOrEmpty(applicationName))
-        {
-            throw new ArgumentNullException(nameof(applicationName));
-        }
+        ArgumentThrowHelper.ThrowIfNullOrEmpty(applicationName);
```

Comment:
is it OK that these are theoretical breaking changes in the empty string case, changing the exception type?

Reply:
(I assume so, I believe dotnet/runtime does such things.)

Reply:
This is true, although I'm comfortable documenting this as a breaking change and explaining.